filebeat/harvester/log.go

```diff
@@ -44,13 +44,18 @@ func (h *Harvester) Harvest() {
 
 	logp.Info("Harvester started for file: %s", h.state.Source)
 
-	r, err := h.newLogFileReader()
-
+	h.reader, err = h.newLogFileReader()
 	if err != nil {
 		logp.Err("Stop Harvesting. Unexpected encoding line reader error: %s", err)
 		return
 	}
 
+	// Waits for harvester channel to be closed to close reader
+	go func() {
+		<-h.done
+		h.reader.Close()
```

Comment:
If return is used, no need to close the readers. The close is only to unblock potential hanging readers.